src/chainstate/burn/db/sortdb.rs

```diff
@@ -2254,13 +2397,85 @@ impl <'a> SortitionHandleTx <'a> {
                     keys.push(db_keys::pox_last_anchor().to_string());
                     values.push("".to_string());
                 }
+                // if we've selected an anchor _and_ know of the anchor,
+                //  write the reward set information
+                if let Some(mut reward_set) = reward_info.known_selected_anchor_block_owned() {
+                    if reward_set.len() > 0 {
+                        // if we have a reward set, then we must also have produced a recipient
+                        //   info for this block
+                        let mut recipients_to_remove: Vec<_> = recipient_info.unwrap().recipients.iter().map(
+                            |(addr, ix)| (addr.clone(), *ix)).collect();
+                        recipients_to_remove.sort_unstable_by(|(_, a), (_, b)| b.cmp(a));
```

Comment:
Why is this an unstable sort?

Reply:
Because not only does the ordering of equal indices not matter, the indices themselves should each be unique, so we can use a more efficient unstable sort.